nsxt/resource_nsxt_upgrade_prepare.go

```
@@ -159,6 +159,37 @@ func prepareForUpgrade(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+func isVCF9HostUpgrade(m interface{}, targetVersion string) bool {
+	// In VCF9.0 and newer, overall upgrade status can be PAUSED when hosts are pre-upgraded. Pre-checks should still
+	// execute in this case
+	if util.VersionLower(targetVersion, "9.0.0") {
+		return false
+	}
+	connector := getPolicyConnector(m)
+	client := upgrade.NewStatusSummaryClient(connector)
+	statusSummary, err := client.Get(nil, nil, nil)
+	if err != nil {
+		return false
```

Do you think it might be better to fail here rather than returning false?